src/passes/builtinHandler/explicitConversionToFunc.ts

```diff
@@ -99,24 +99,31 @@ export class ExplicitConversionToFunc extends ASTMapper {
     }
 
     if (typeTo instanceof AddressType) {
-      if (
+      const operand = node.vArguments[0];
+      if (argType instanceof AddressType) {
+        ast.replaceNode(node, operand);
+      } else if (argType instanceof IntLiteralType) {
+        operand.typeString = 'address';
+        ast.replaceNode(node, operand);
+      } else if (
         (argType instanceof IntType && argType.nBits === 256) ||
         (argType instanceof FixedBytesType && argType.size === 32)
       ) {
         const replacementCall = createCallToFunction(
           ast.registerImport(
             node,
-            [...WARPLIB_MATHS, 'utils'],
-            'uint256_to_address_felt',
+            ...UNSAFE_CONTRACT_ADDRESS_FROM_U256,
             [['uint_arg', createUint256TypeName(ast)]],
             [['address_ret', createAddressTypeName(false, ast)]],
           ),
-          [node.vArguments[0]],
+          [operand],
           ast,
         );
         ast.replaceNode(node, replacementCall);
       } else {
-        ast.replaceNode(node, node.vArguments[0]);
+        throw new TranspileFailedError(
+          `Don't know how to convert ${argType.constructor.name} to address`,
```

Comment:
Unknown type `${argType.constructor.name}` received for addressType conversion

Reply:
Oh god, took me a minute to understand that you weren't describing a warp error :D 

Will change